src/Stream/ReadableStreamTrait.php

```diff
@@ -102,6 +102,9 @@ protected function performRead($chunkSize)
                 $this->emit('end', [
                     $this,
                 ]);
+                $this->emit('close', [
+                    $this,
+                ]);
```

Comment:
Makes sense. Maybe call `close()` instead here? Could also use a test or two :+1:

Reply:
Replacing both lines with close and calling close instead. Will also add tests to be sure everything works as intended :+1:

Reply:
Solved the issue, working on tests now. (It required some other related code change as well.)

Reply:
PR updated